### PR TITLE
Updating, and adding docs and code cleanup

### PR DESCRIPTION
**Pull request by briedel**

### CODE REVIEW

File: client.py [hunk 1]
```diff
@@ -156,6 +174,9 @@ def main():
 
                 if "count" in s and s["count"] > limit:
                     s["count"] = limit
+                if ("max_memory_per_job" in config_cluster 
+                    and s["memory"] > config_cluster["max_memory_per_job"]):
+                    continue
```
Comment (dsschult) [inline client.py]: These three lines aren't needed anymore. They are covered by the for loop above.

File: client_util.py [hunk 2]
```diff
@@ -92,9 +94,19 @@ def monitoring(address,info=None):
...
+    Parsing config file
+
+    Args:
+        config: Pythong config parser object
+
+    Returns:
+        A dict with the different sections of the config file
+        and the literal values of the configuraton objects
+    """
+    config_dict = dict()
```
Comment (dsschult) [inline client_util.py]: Why `dict()` instead of `{}`?

File: submit.py [hunk 3]
```diff
@@ -172,7 +188,8 @@ def write_glidein_part(self, f, local_dir=None, glidein_tarball=None,
         if "CustomEnv" in self.config:
             for k, v in self.config["CustomEnv"].items():
                 f.write(k + '=' + v + ' ')
-        f.write('./%s\n' % glidein_script)
+        self.write_line(f, 'cd $LOCAL_DIR')
```
Comment (dsschult) [inline submit.py]: There's already a `cd` several lines up.  And you can't put that inside the `env -i` command.

File: client_util.py (hunk 2 above)
Comment (briedel) [inline client_util.py]: `{}` could also be a set

Comment (dsschult) [inline client_util.py]: From the python docs:

> Note: to create an empty set you have to use set(), not {}; the latter creates an empty dictionary, a data structure that we discuss in the next section.

Comment (dsschult) [inline client_util.py]: Also note this article:
https://doughellmann.com/blog/2012/11/12/the-performance-impact-of-using-dict-instead-of-in-cpython-2-7-2/

File: client.py [hunk 4]
```diff
@@ -153,16 +153,16 @@ def main():
                 if limit <= 0:
                     logger.info('reached limit')
                     break
+                skip = False
                 # Skipping CPU jobs for gpu only clusters
                 if ('gpu_only' in config_cluster and config_cluster['gpu_only']
                     and s["gpus"] == 0):
-                    continue
+                    skip = True
```
Comment (dsschult) [inline client.py]: I liked `continue` here more because it short-circuited the for loop.

**Pull request merged**